src/Bicep.LangServer/BicepCompilationManager.cs

```diff
@@ -231,11 +250,43 @@ private ImmutableArray<ISourceFile> CloseCompilationInternal(DocumentUri documen
                     Code = new DiagnosticCode("Fatal")
                 };
 
-                // the file is no longer in a state that can be parsed
-                // clear all info to prevent cascading failures elsewhere
-                var closedFiles = CloseCompilationInternal(documentUri, version, fatalError.AsEnumerable());
+                this.PublishDocumentDiagnostics(documentUri, version, fatalError.AsEnumerable());
+
+                return (ImmutableArray<ISourceFile>.Empty, ImmutableArray<ISourceFile>.Empty);
+            }
+        }
+
+        private void ReloadBicepConfig(DocumentUri documentUri)
+        {
+            var folderContainingBicepFile = Path.GetDirectoryName(documentUri.GetFileSystemPath());
+
+            if (folderContainingBicepFile is not null)
+            {
+                string localBicepConfig = Path.Combine(folderContainingBicepFile, LanguageConstants.BicepConfigSettingsFileName);
+                BicepConfig? bicepConfig = null;
+
+                // If bicepconfig.json exists in the folder containing the source files, we'll use local config settings.
+                // Otherwise use settings from parent directory(if bicepconfig.json is found in one of parent directories) or default settings
+                if (File.Exists(localBicepConfig))
```

Comment:
In general, avoid using `File.<something>()` methods - instead use the `fileResolver` instance. This allows us to mock out the filesystem in unit tests if we want to.

Reply:
Don't need the check anymore. Updated to use ConfigHelper.DiscoverLocalConfigurationFile(..) here:
https://github.com/Azure/bicep/commit/bbb280e8f2fe05d4c7000044b10e5f9715fcb854